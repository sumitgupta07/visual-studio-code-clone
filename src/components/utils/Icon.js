import React from 'react';
import './Icon.css';

const Icon = ({ label, onSelectIcon, iconClass }) => {
  return (
    <button onClick={() => onSelectIcon(label)}>
      <div className="tooltip">
        <div className="icon">{<i className={iconClass}></i>}</div>
        <div className="tooltiptext">
          <span>{label}</span>
        </div>
      </div>
    </button>
  );
};

export default Icon;
