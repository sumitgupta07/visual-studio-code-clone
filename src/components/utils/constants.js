export const labels = {
  FileExplorer: 'File Explorer',
  Search: 'Search',
  SourceControl: 'Source Control',
  Debug: 'Debug',
  Extensions: 'Extensions'
};

export const classNames = {
  explorerClass: 'far fa-copy fa-3x',
  searchClass: 'fas fa-search fa-rotate fa-3x',
  sourcecontrolClass: 'fas fa-code-branch fa-3x',
  debugClass: 'fas fa-bug fa-3x',
  extensionsClass: 'fas fa-th-large fa-3x'
};

export const folders = [
  {
    id: 1,
    name: 'src',
    files: [
      {
        id: 'a',
        fileName: 'index.js',
        content: 'Placeholder for index.js. It has App.js as child components'
      },
      {
        id: 'b',
        fileName: 'App.js',
        content:
          'Placeholder for App.js. It has Header and Content as child components'
      },
      {
        id: 'c',
        fileName: 'Content.js',
        content:
          'Placeholder for Content.js file. It has Navbar, SelectPane and EditorPane as child components'
      }
    ]
  }
];
