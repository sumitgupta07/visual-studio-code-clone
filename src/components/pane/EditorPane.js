import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { folders } from '../utils/constants';
import FileTab from '../Editor/FileTab';
import FileContent from '../Editor/FileContent';

const EditorPane = ({ activeFile, onSelectFile }) => {
  const [currentFile, setCurrentFile] = useState({});

  //This observes changes in state of File Explorer component
  useEffect(() => {
    setCurrentFile(activeFile);
  }, [activeFile]);

  //This sets current selected file in Editor Pane & File Explorer
  const onClickFileItem = file => {
    setCurrentFile(file);
    onSelectFile(file);
  };
  return (
    <div className="editorpane">
      <h3>Editor Pane</h3>
      <div className="files">
        <ul className="file-list">
          {folders[0].files.map(file => (
            <FileTab
              activeFile={currentFile}
              key={file.id}
              file={file}
              onClick={onClickFileItem}
            />
          ))}
        </ul>
      </div>
      {folders[0].files.map(file => {
        if (file.id === currentFile.id)
          return <FileContent key={currentFile.id} currentFile={currentFile} />;
        else return null;
      })}
    </div>
  );
};

EditorPane.propTypes = {
  activeFile: PropTypes.object.isRequired
};

export default EditorPane;
