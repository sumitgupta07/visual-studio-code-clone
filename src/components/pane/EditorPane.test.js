import React from 'react';
import { shallow, mount } from 'enzyme';
import EditorPane from './EditorPane';
import { folders } from '../utils/constants';

let wrapper;
let mockProps;
beforeEach(() => {
  mockProps = {
    activeFile: {
      id: 'a',
      fileName: 'index.js',
      content: 'Placeholder for index.js. It has App.js as child components'
    }
  };
  wrapper = shallow(<EditorPane {...mockProps} />);
});

it('should render snapshot of Editor Pane Component', () => {
  expect(wrapper).toMatchSnapshot();
});

it('should render all tab items for all files of first folder', () => {
  expect(wrapper.find('FileTab').length).toEqual(folders[0].files.length);
});

it('should render active file content', () => {
  //wrapper re-assigned to search deep in the Content component
  wrapper = mount(<EditorPane {...mockProps} />);
  expect(wrapper.find('FileContent').text()).toEqual(
    mockProps.activeFile.content
  );
});
