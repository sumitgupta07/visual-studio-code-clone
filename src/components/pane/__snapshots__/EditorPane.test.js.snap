// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render snapshot of Editor Pane Component 1`] = `
<div
  className="editorpane"
>
  <h3>
    Editor Pane
  </h3>
  <div
    className="files"
  >
    <ul
      className="file-list"
    >
      <FileTab
        activeFile={Object {}}
        file={
          Object {
            "content": "Placeholder for index.js. It has App.js as child components",
            "fileName": "index.js",
            "id": "a",
          }
        }
        key="a"
        onClick={[Function]}
      />
      <FileTab
        activeFile={Object {}}
        file={
          Object {
            "content": "Placeholder for App.js. It has Header and Content as child components",
            "fileName": "App.js",
            "id": "b",
          }
        }
        key="b"
        onClick={[Function]}
      />
      <FileTab
        activeFile={Object {}}
        file={
          Object {
            "content": "Placeholder for Content.js file. It has Navbar, SelectPane and EditorPane as child components",
            "fileName": "Content.js",
            "id": "c",
          }
        }
        key="c"
        onClick={[Function]}
      />
    </ul>
  </div>
</div>
`;
